src/a11y-dialog.ts

```diff
@@ -5,6 +5,8 @@ export type A11yDialogInstance = InstanceType<typeof A11yDialog>
 
 const SCOPE = 'data-a11y-dialog'
 
+let closeWatcher: CloseWatcher | null = null
```

Comment:
First thing I don’t love about the `CloseWatcher` API is that it’s single-use.

Once the watcher received a close event, it no longer processes anything else. That means we need to instantiate a new watcher every time we open the dialog. 

There is also no way to know from an existing instance whether or not it’s been consumed or if it can process a close event. That’s not exposed.

Reply:
Yeah I agree the API shape is a bit weird. I guess it makes sense that you don't want to constantly keep listening for a close request.

Does this cause an actual problem?